src/canmatrix/canmatrix.py

```diff
@@ -436,25 +436,27 @@ def phys2raw(self, value=None):
             if not (self.min <= value <= self.max):
                 value = self.min
 
-        if isinstance(value, str):
+        if isinstance(value, str) and self.values:
             for value_key, value_string in self.values.items():
                 if value_string == value:
                     value = value_key
                     return value
-            else:
-                raise ValueError(
-                        "{} is invalid value choice for {}".format(value, self)
-                )
 
-        if not (self.min <= value <= self.max):
-            logger.info(
+        try:
+            value = decimal.Decimal(value)
+        except Exception as e:
+            raise e
+
+        if not (0 <= value <= 10):
```

Comment:
Hi @xRowe , why 0 to 10 and not self.min and self.max?

Reply:
Test Code...
Found Issue. To be revert to normal code